specfile/specfile.py

```diff
@@ -120,18 +183,32 @@ def _dump_debug_info(self, message) -> None:
         )
 
     @staticmethod
-    def _read_lines(path: Path) -> Tuple[List[str], bool]:
-        content = path.read_text(encoding="utf8", errors="surrogateescape")
-        return content.splitlines(), content[-1] == "\n"
+    def _read_lines(stream: TextIO) -> Tuple[List[str], bool]:
+        stream.seek(0)
+        content = stream.read()
+        return content.splitlines(), content.endswith("\n")
+
+    @property
+    def is_writable(self) -> bool:
+        return hasattr(self._file, "write") and callable(
+            getattr(self._file, "write", None)
+        )
 
     @property
     def path(self) -> Path:
         """Path to the spec file."""
-        return self._path
+        if hasattr(self._file, "name"):
+            return Path(self._file.name)
+        return None
 
     @path.setter
     def path(self, value: Union[Path, str]) -> None:
-        self._path = Path(value)
+        self._file.close()
+        new_path = Path(value)
+        self._file = new_path.open("r+", encoding="utf8", errors="surrogateescape")
+        if not self.is_sourcedir_set:
+            self._parser.sourcedir = new_path.parent
+        self.reload()
```

Comment:
I wouldn't do this, the original implementation also didn't change sourcedir when changing path. Even if a user didn't specify sourcedir explicitly (which you can't possibly know for sure from constructor arguments, the user could use the sourcedir property to do that), they may want to keep the sources where they are and just move the spec file elsewhere. It is true though that we should document this - if they change path sourcedir won't be automatically updated even if they didn't specify it explicitly.